### PR TITLE
Emit progid information into the CLSIDMap.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This is the SDK-side of the work for https://github.com/dotnet/core-setup/pull/7551. These PRs allow users to activate .NET classes via COM using ProgIDs instead of CLSIDs for languages or platforms that do not support using CLSIDs.

Additionally, fixes a bug that would cause loading nested classes via COM to fail to find the type.

Contributes to dotnet/coreclr#25966

This PR doesn't need to wait for dotnet/core-setup#7551 to go in since the changes are backward compatible with previous versions of the COM host that don't have ProgID support.

cc: @nguerrera 

================================

#### Description
Add support for generating .cslidmaps with ProgIDs for com registration. See dotnet/core-setup#7573 for the host work. This PR also fixes a bug that breaks loading nested types via COM.

#### Customer Impact
Without this support consumption of COM classes in languages that don't have `GUID` support is impossible.

Additionally, without the fix for nested types, it is impossible to activate a nested type via COM on .NET Core.

Fixes: dotnet/coreclr#25966
Related: https://github.com/dotnet/coreclr/issues/25946

#### Regression?
COM support is new in 3.0 so this is not a regression from a previous .NET Core release, but is a regression in COM support from .NET Framework using the `RegAsm` tool.

#### Risk
Minimal. Unused information in the clsidmap such as ProgId information is ignored in previous previews of .NET Core.

### CODE REVIEW

**Comment by nguerrera** [comment]
What release is this for?

**Comment by jkoritzinsky** [comment]
This needs to go into 3.0.

File: src/Tasks/Microsoft.NET.Build.Tasks/ClsidMap.cs
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/ClsidMap.cs]: Umm

Comment (nguerrera) [reply src/Tasks/Microsoft.NET.Build.Tasks/ClsidMap.cs]: ;)

Comment (jkoritzinsky) [reply src/Tasks/Microsoft.NET.Build.Tasks/ClsidMap.cs]: Thanks. Totally forgot to remove that.

**Comment by nguerrera** [comment]
At some point, I would like to see this code moved to HostModel in core-setup.

**Comment by nguerrera** [comment]
cc @livarcocc

**Comment by jkoritzinsky** [comment]
That's the plan for after 3.0.

**Comment by nguerrera** [comment]
> This needs to go into 3.0.

And you have approval for that? 

We don't dual checkin master and 3.0 in SDK, please retarget to release/3.0.1xx

**Comment by jkoritzinsky** [comment]
We're still waiting on approval. Do you want me to re-target for release/3.0.1xx pending approval or review for master for now and then release/3.0.1xx after approval?

**Comment by livarcocc** [comment]
@jkoritzinsky please re-target. Also fill the ask mode template and mark this bug with ask-mode label so that it shows up in the approval query for preview9.

**Comment by jkoritzinsky** [comment]
@livarcocc I don't have permissions to add the ask-mode label. Can you add the label while I fill out the template?

**Comment by jkoritzinsky** [comment]
@nguerrera @livarcocc I've filled out the Ask Mode template. Can one of you add the `ask-mode` label to this since I don't have permissions to do so on the SDK repo?

**Review by nguerrera — APPROVED** [review]

**Comment by nguerrera** [comment]
Done

**Comment by AaronRobinsonMSFT** [comment]
@nguerrera This has been approached from the CoreCLR point of view. Is there anything else that is needed?

**Comment by nguerrera** [comment]
> @nguerrera This has been approached from the CoreCLR point of view. Is there anything else that is needed?

I don't think anything else is needed on our side. I'm relying on your team to tell us if there is. This doesn't appear any different than the other processing we have in the SDK.

**Comment by AaronRobinsonMSFT** [comment]
@nguerrera This was approved by ship room.

/cc @jeffschwMSFT

**Pull request merged**
